src/draw/lv_draw_vector.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file lv_draw_vector.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_draw_vector.h"
+#include "../stdlib/lv_mem.h"
+#include "../stdlib/lv_string.h"
+#include <math.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#ifndef M_PI
+    #define M_PI 3.1415926f
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC FUNCTIONS
+ **********************/
+
+static inline void _matrix_translate(float m[3][3], float x, float y)
```

Comment:
Let's move to header files in case widgets may need it?

Reply:
> Let's move to header files in case widgets may need it?

I think we should add a set of matrix api like  
lv_matrix_translate
lv_matrix_scale
.....
Because some GPUs can provide the interface of matrix transformation 
For example:
               vg_lite_rotate
               vg_lite_scale
               ....
Or we can also acceleration at the bottom with  NEON Instrinsic.

Reply:
I agree with bringing the declarations to public, however since vg_lite matrix APIs are naive matrix multiplications I would advice against using them.